avm/res/sql/managed-instance/main.bicep

```diff
@@ -0,0 +1,527 @@
+param name string
+
+@description('Optional. Location for all resources.')
+param location string = resourceGroup().location
+
+@description('Required. The username used to establish jumpbox VMs.')
+param administratorLogin string
+
+@description('Required. The password given to the admin user.')
+@secure()
+param administratorLoginPassword string
+
+@description('Required. The fully qualified resource ID of the subnet on which the SQL managed instance will be placed.')
+param subnetId string
+
+@description('Optional. The name of the SKU, typically, a letter + Number code, e.g. P3.')
+param skuName string = 'GP_Gen5'
+
+@description('Optional. The tier or edition of the particular SKU, e.g. Basic, Premium.')
+param skuTier string = 'GeneralPurpose'
+
+@description('Optional. Storage size in GB. Minimum value: 32. Maximum value: 8192. Increments of 32 GB allowed only.')
+param storageSizeInGB int = 32
+
+@description('Optional. The number of vCores. Allowed values: 8, 16, 24, 32, 40, 64, 80.')
+param vCores int = 4
+
+@description('Optional. The license type. Possible values are \'LicenseIncluded\' (regular price inclusive of a new SQL license) and \'BasePrice\' (discounted AHB price for bringing your own SQL licenses).')
+@allowed([
+  'LicenseIncluded'
+  'BasePrice'
+])
+param licenseType string = 'LicenseIncluded'
+
+@description('Optional. If the service has different generations of hardware, for the same SKU, then that can be captured here.')
+param hardwareFamily string = 'Gen5'
+
+@description('Optional. Whether or not multi-az is enabled.')
+param zoneRedundant bool = false
+
+@description('Optional. Service principal type. If using AD Authentication and applying Admin, must be set to `SystemAssigned`. Then Global Admin must allow Reader access to Azure AD for the Service Principal.')
+@allowed([
+  'None'
+  'SystemAssigned'
+])
+param servicePrincipal string = 'None'
+
+@description('Optional. Specifies the mode of database creation. Default: Regular instance creation. Restore: Creates an instance by restoring a set of backups to specific point in time. RestorePointInTime and SourceManagedInstanceId must be specified.')
+@allowed([
+  'Default'
+  'PointInTimeRestore'
+])
+param managedInstanceCreateMode string = 'Default'
+
+@description('Optional. The resource ID of another managed instance whose DNS zone this managed instance will share after creation.')
+param dnsZonePartner string = ''
+
+@description('Optional. Collation of the managed instance.')
+param collation string = 'SQL_Latin1_General_CP1_CI_AS'
+
+@description('Optional. Connection type used for connecting to the instance.')
+@allowed([
+  'Proxy'
+  'Redirect'
+  'Default'
+])
+param proxyOverride string = 'Proxy'
+
+//@description('Optional. Configuration details for private endpoints. For security reasons, it is recommended to use private endpoints whenever possible.')
+//param privateEndpoints privateEndpointType
```

Comment:
Why is this commented? Are Private Endpoints not supported? And if not, then this, as well as the UDT, should be removed

Reply:
SQL MI does support private endpoints. I imagine that this was commented out as a placeholder to put them in the code; now that they are supported. We can remove the comment for now and then work on adding private endpoints after the module is migrated from CARML to AVM. Currently, CARML does not have them included.

Reply:
Yey the comment is nowhere to be seen in CARML 😄 . Will resolve regardless